File: Final-updated-job-portel/OnlineJobPortal/src/main/java/com/coforge/service/SkillService.java
package com.coforge.service;

import java.util.List;

import com.coforge.entity.Skill;

public interface SkillService {

    Skill createSkill(Skill skill);

    Skill getSkillById(Long id);

    List<Skill> getAllSkills();

    void deleteSkill(Long id);
}

File: Final-updated-job-portel/frontend/src/app/services/skill.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { Skill } from '../models/api.models';

@Injectable({ providedIn: 'root' })
export class SkillService {
  constructor(private readonly api: ApiService) {}

  getAll(): Observable<Skill[]> {
    return this.api.getSkills();
  }

  create(name: string): Observable<Skill> {
    return this.api.createSkill({ name });
  }

  remove(id: number): Observable<void> {
    return this.api.deleteSkill(id);
  }
}
